src/Event/Events/EventFromUDB2.php

```diff
@@ -201,4 +215,28 @@ public function getExternalId(): ?string
         $eventAsArray = $this->getEventAsArray();
         return $eventAsArray['location'][0]['label'][0]['@attributes']['externalid'] ?? null;
     }
+
+    private function getModeration(?string $wfstatus, ?string $lastUpdated): ?AbstractEvent
+    {
+        if ($wfstatus === 'readyforvalidation') {
+            return new Published(
+                $this->eventId,
+                \DateTimeImmutable::createFromFormat('Y-m-d\TH:i:s', $lastUpdated, new DateTimeZone('Europe/Brussels')),
+            );
+        }
+
+        if ($wfstatus === 'approved') {
+            return new Approved($this->eventId);
+        }
+
+        if ($wfstatus === 'rejected') {
+            return new Rejected($this->eventId, new StringLiteral('Reason unknown (imported from UiTdatabank v2)'));
+        }
+
+        if ($wfstatus === 'deleted') {
+            return new EventDeleted($this->eventId);
+        }
+
+        return null;
```

Comment:
Which `wfstatus` in XML can not be converted to a moderation event?

Reply:
The `wfstatus` could still be `DRAFT` in `CdbXml`, which is not a moderation event in UDB3.

Reply:
Thanks for clarifying, maybe add a small comment for this in the code?

Reply:
Done.